ingest-app/src/cmr/ingest/services/email_processing.clj

```diff
@@ -0,0 +1,251 @@
+   [cmr.transmit.access-control :as access-control]
+   [cmr.transmit.config :as config]
+   [cmr.transmit.metadata-db :as mdb]
+   [cmr.transmit.search :as search]
+   [markdown.core :as markdown]
+   [postal.core :as postal-core]))
+
+;; Call the following to trigger a job, example below will fire an email subscription
+;; UPDATE QRTZ_TRIGGERS
+;; SET NEXT_FIRE_TIME =(((cast (SYS_EXTRACT_UTC(SYSTIMESTAMP) as DATE) - DATE'1970-01-01')*86400 + 1200) * 1000)
+;; WHERE trigger_name='EmailSubscriptionProcessing.job.trigger';
+
+;; Specs =============================================================
+(def date-rx "\\d{4}-\\d{2}-\\d{2}T\\d{2}:\\d{2}:\\d{2}\\.\\d{3}Z")
+
+(def time-constraint-pattern (re-pattern (str date-rx "," date-rx)))
+
+(spec/def ::time-constraint (spec/and
+                              string?
+                              #(re-matches time-constraint-pattern %)))
+
+(defconfig email-subscription-processing-lookback
+  "Number of seconds to look back for granule changes."
+  {:default 3600
+   :type Long})
+
+(defconfig email-server-host
+  "The host name for email server."
+  {:default ""
+   :type String})
+
+(defconfig email-server-port
+  "The port number for email server."
+  {:default 25
+   :type Long})
+
+(defconfig mail-sender
+  "The email sender's email address."
+  {:default ""
+   :type String})
+
+(def subscription-permission-notification-expiration-days
+  "The number of days to wait before deleting subscription and notifying user that their permission
+   to view the collection was removed"
+   3)
+
+(defn- create-query-params
+  "Create query parameters using the query string like
+  \"polygon=1,2,3&concept-id=G1-PROV1\""
+  [query-string]
+  (let [query-string-list (string/split query-string #"&")
+        query-map-list (map #(let [a (string/split % #"=")]
+                               {(first a) (second a)})
+                             query-string-list)]
+     (apply merge query-map-list)))
+
+(defn email-granule-url-list
+ "take a list of URLs and format them for an email"
+ [gran-ref-location]
+ (string/join "\n" (map #(str "* [" % "](" % ")") gran-ref-location)))
+
+(defn failed-permission-content
+  "Create an email body for failed permission notifications."
+  [from-email-address to-email-address coll-id]
+  {:from from-email-address
+   :to to-email-address
+   :subject "You can no longer view this collection"
+   :body [{:type "text/html"
+           :content (markdown/md-to-html-string
+                     (str "You are currently subscribed to receive updates for new data added to " coll-id ".\n\n"
+                          "Because you no longer have access to view this collection, you will not receive notifications "
+                          "related to this collection.\n\n"
+                          "If you believe this subscription was deleted in error, please contact us at [cmr-support@earthdata.nasa.gov](mailto:cmr-support@earthdata.nasa.gov).\n\n"))}]})
```

Comment:
I don't see we delete the subscription in the code. This message needs to be updated.